app/frontend/src/common/store/documents.js

```diff
@@ -111,7 +114,23 @@ export default {
       state.file_upload_success = payload
     },
     setFiles (state, payload) {
-      state.upload_files = payload
+      if (payload.length > 0) {
+        payload.forEach((file) => {
+          state.upload_files.push(file)
+        })
+      } else {
+        state.upload_files = payload
+      }
+    },
+    removeFile (state, file) {
+      let l = state.upload_files.length
```

Comment:
Even simpler - one line - but don't know of replacing the array will work for you?
```suggestion
      state.upload_files = state.upload_files.filter((item) => item.name === file)
```